server/src/org/apache/cloudstack/annotation/AnnotationManagerImpl.java

```diff
@@ -0,0 +1,145 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package org.apache.cloudstack.annotation;
+
+import com.cloud.utils.component.ManagerBase;
+import com.cloud.utils.component.PluggableService;
+import org.apache.cloudstack.annotation.dao.AnnotationDao;
+import org.apache.cloudstack.api.command.admin.annotation.AddAnnotationCmd;
+import org.apache.cloudstack.api.command.admin.annotation.ListAnnotationsCmd;
+import org.apache.cloudstack.api.command.admin.annotation.RemoveAnnotationCmd;
+import org.apache.cloudstack.api.response.AnnotationResponse;
+import org.apache.cloudstack.api.response.ListResponse;
+import org.apache.cloudstack.context.CallContext;
+import org.apache.log4j.Logger;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @since 4.11
+ */
+public final class AnnotationManagerImpl extends ManagerBase implements AnnotationService, PluggableService {
+    public static final Logger s_logger = Logger.getLogger(AnnotationManagerImpl.class);
+
+    @Inject
+    private AnnotationDao annotationDao;
+
+    @Override
+    public ListResponse<AnnotationResponse> searchForAnnotations(ListAnnotationsCmd cmd) {
+        List<AnnotationVO> annotations =
+                getAnnotationsForApiCmd(cmd);
+        List<AnnotationResponse> annotationResponses =
+                convertAnnotationsToResponses(annotations);
+        return createAnnotationsResponseList(annotationResponses);
+    }
+
+    @Override
+    public AnnotationResponse addAnnotation(AddAnnotationCmd addAnnotationCmd) {
+        return addAnnotation(addAnnotationCmd.getAnnotation(), addAnnotationCmd.getEntityType(), addAnnotationCmd.getEntityUuid());
+    }
+
+    public AnnotationResponse addAnnotation(String text, EntityType type, String uuid) {
```

Comment:
General comment -- the API handling methods can use the events annotations that are used to create events when APIs are called. See other API such as (dynamic) roles APIs for example.